packages/components/text/src/Text.tsx

```diff
@@ -106,6 +113,19 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
     ['color', 'fontStyle', 'textAlign', 'textDecorationLine', ...fontStyles.keys],
   );
 
+  // [TODO(#2268): Remove once RN Core properly supports Dynamic Type scaling
+  let scaleStyleAdjustments: TextTokens = emptyProps;
+  // tokenStyle.fontSize and tokenStyle.lineHeight can also be strings (e.g., "14px").
+  // Therefore, we only support scaling for number-based size values in order to avoid any messy calculations.
+  if (dynamicTypeVariant !== undefined && typeof tokenStyle.fontSize === 'number' && typeof tokenStyle.lineHeight === 'number') {
+    const scaleFactor = fontMetricsScaleFactors[dynamicTypeVariant] ?? 1;
+    scaleStyleAdjustments = {
+      fontSize: tokenStyle.fontSize * scaleFactor,
+      lineHeight: tokenStyle.lineHeight * scaleFactor,
+    };
+  }
+  // ]TODO(#2268)
```

Comment:
To make it more clear this is an iOS only calculation that should be done, can we put this inside a `if (Platform.OS === 'ios')` check? I try to avoid platform specific checks, but this feels like a valid case. I know that the scale adjustments would just return "1", but I don't think that's obvious if someone is skimming this file. In a similar vein, could we add a comment to this block of code to the vein of `"Make iOS specific adjustments to the font scaling to better match Apple's Dynamic Type ramp"`?

Reply:
I made the `Platform.OS == 'ios'` check as part of defining `dynamicTypeVariant` because we also check the value of `dynamicTypeVariant` in the spread that determines whether or not we should add `{allowFontScaling: false}`.